### PR TITLE
Testing the 'generator' module.

### PR DESCRIPTION
**Pull request by jsanemet**
As this module is still not definitive and is bound to change, I have avoided testing the 'main' function yet. Coverage will not be as high because this only covers testing of the surrounding functions. Additionally, I have added a small fix to have the tests pass correctly.

### CODE REVIEW

File: tests/test_generator.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+
+class TestFetchOSPProjects(TestCase):
+    TEMPLATES_REPOSITORY = 'https://opendev.org/openstack/openstack-zuul-jobs'
+    PACKAGE_REPOSITORY = 'http://localhost:8080/my/package'
+
+    TEMPLATES_DIR = 'openstack/openstack-zuul-jobs/zuul.d'
+    PACKAGE_DIR = 'my/package'
+
+    def download_zuul_config(self, **kwargs):
+        match kwargs.get('repository'):
```
Comment (cescgina) [inline tests/test_generator.py:48]: Isn't the match statment new to python3.10? I think it's too soon to rely on it, our CI for example runs on py39

Comment (jsanemet) [reply tests/test_generator.py:48]: I had not noticed, I will change it for some 'if...else' then.

Comment (sdatko) [reply tests/test_generator.py]: +1, I prefer traditional if/else, match is too new IMO

Comment (jsanemet) [reply tests/test_generator.py]: I have uploaded a change removing the 'match' statement, this should now be compatible with Python 3.9.

**Review by bregman-arie — APPROVED** [review]
Really liked this change

**Review by cescgina — APPROVED** [review]
Looks great!

**Pull request merged**